compiler/rustc_passes/src/dead.rs

```diff
@@ -274,10 +272,9 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
         if let Some(item_def_id) = match node {
             Node::ImplItem(hir::ImplItem { def_id, .. }) => Some(def_id.to_def_id()),
             _ => None,
-        } {
-            if self.should_ignore_item(item_def_id) {
-                return;
-            }
+        } && self.should_ignore_item(item_def_id)
+        {
+            return;
         }
```

Comment:
this change is arguably harder to read/understand...

Reply:
if you are going to change this, then please refactor it further, like:

```
match node {
  Node::ImplItem(..) if should_ignore() => return,
}
```

Reply:
I fixed it.